src/Model/Item.php

```diff
@@ -201,18 +201,10 @@ public static function selectPinned(int $uid, array $selected = [], array $condi
 			return [];
 		}
 
-		if (empty($condition) || !is_array($condition)) {
-			$condition = ['iid' => $pinned];
+		if (!empty($condition)) {
+			$condition = DBA::mergeConditions(['iid' => $pinned], $condition);
 		} else {
-			reset($condition);
-			$first_key = key($condition);
-			if (!is_int($first_key)) {
-				$condition['iid'] = $pinned;
-			} else {
-				$values_string = substr(str_repeat("?, ", count($pinned)), 0, -2);
-				$condition[0] = '(' . $condition[0] . ") AND `iid` IN (" . $values_string . ")";
-				$condition = array_merge($condition, $pinned);
-			}
+			$condition = ['iid' => $pinned];
 		}
```

Comment:
You can merge a condition with an empty array and replace this whole block with the following:
```php
$condition = DBA::mergeConditions($condition, ['iid' => $pinned]);
```

Reply:
It is some (slight) performance advantage not calling that function, since it doesn't have any checks for this condition.

Reply:
I don't think we are at that level of optimization, I'd rather reduce the number of conditions and code lines in general.